packages/headless/src/controllers/facets/range-facet/date-facet/date-range.test.ts

```diff
@@ -0,0 +1,91 @@
+import {DateRangeRequest} from '../../../../features/facets/range-facets/date-facet-set/interfaces/request';
```

Comment:
These tests are mostly as they were in `headless-date-facet.test.ts`. I extracted them into a new file, and added tests for a new exported function, `isSearchApiDate`